xikolo-ios/Helpers/AnalyticsHelper.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  AnalyticsHelper.swift
+//  xikolo-ios
+//
+//  Created by Bjarne Sievers on 27.08.17.
+//  Copyright © 2017 HPI. All rights reserved.
+//
+
+import Foundation
+
+class AnalyticsHelper {
+
+    class func setup() {
+        NotificationCenter.default.addObserver(self, selector: #selector(AnalyticsHelper.check), name: NotificationKeys.reachabilityChanged, object: nil)
+        check()
+    }
+
+    @objc class func check() {
+        do {
+            let eventCount = try CoreDataHelper.backgroundContext.count(for: TrackingHelper.getTrackingEventRequest())
+            if eventCount > 0 {
+                self.uploadTrackingEvents()
+            }
+        } catch {
+            print("\(error)")
+        }
+    }
+
+    class func uploadTrackingEvents() {
+        guard ReachabilityHelper.reachability.currentReachabilityStatus == .reachableViaWiFi else { return }
```

Comment:
we mights send via mobile network as well

Reply:
But why? I wouldn't want this as a user. What's the benefit for us? Since there is a queue now, everything should still get sent eventually.

Reply:
Because you might never be online via WLAN. On Android we decide this per event, some unimportant are wifi only, other sync as soon as possible